Show and hide menu windows once visibility is changed

msisov opened the pull request:
This cl adds Show and Hide API to the PlatformDisplayDefault
so that ones menus change they visibility and request hide
or show, native windows are shown or hidden

tonikitoo commented on services/ui/ws/window_server.cc:
```diff
@@ -751,9 +751,25 @@ void WindowServer::OnWindowVisibilityChanged(ServerWindow* window) {
+  }
+}
+
+void WindowServer::OnHideNativeWindow(ServerWindow* window) {
```
can not this be merged with the OnWindowVisibilityChanged method above?

It would make the diff smaller in a couple of places

msisov replied:
Nope, if we hide window on each visibilty changed to false, main window will just disappear. That's the reason why I have it separately from visibility.

tonikitoo commented on services/ui/ws/window_tree.cc:
```diff
@@ -1566,6 +1566,14 @@ void WindowTree::SetWindowVisibility(uint32_t change_id,
       SetWindowVisibility(ClientWindowId(transport_window_id), visible));
 }
 
+void WindowTree::HideNativeWindow(Id transport_window_id) {
```
maybe add a DCHECK for external window mode?

msisov replied:
it goes here from WindowTreeHostMus::HideImpl. In internal window mode, it will produce a crash

msisov committed:
Show and hide menu windows once visibility is changed

This cl adds Show and Hide API to the PlatformDisplayDefault
so that ones menus change they visibility and request hide
or show, native windows are shown or hidden

tonikitoo commented on services/ui/ws/platform_display.h:
```diff
@@ -61,6 +61,10 @@ class PlatformDisplay : public ui::EventSource {
   virtual FrameGenerator* GetFrameGenerator() = 0;
 
+  // Shows or hides native window.
+  virtual void Hide() {}
```
These are the first occurrence of "Show" in services/ui/ws.

Maybe stick with the SetWindowVisibility-like nomenclature?

tonikitoo commented on services/ui/ws/window_server.cc:
```diff
@@ -751,9 +751,20 @@ void WindowServer::OnWindowVisibilityChanged(ServerWindow* window) {
+    if (IsInExternalWindowMode()) {
+      PlatformDisplay* display = display_root->display()->platform_display();
+      DCHECK(display);
+      if (window->visible()) {
```
nit: maybe eliminate {} for these one line clauses?

tonikitoo commented on services/ui/ws/event_dispatcher.cc:
```diff
@@ -400,7 +400,17 @@ void EventDispatcher::ProcessPointerEvent(const ui::PointerEvent& event) {
+// it as long as capture is already set by the main window. This breaks
+// part functionality of menu windows.
+// TODO(msisov, tonikitoo): figure out this.
+#if defined(OS_CHROMEOS)
```
we are using

#if defined(OS_CHROMEOS) || !defined(USE_OZONE)

because Mus is also available for windows and other google configurations.

msisov replied:
done!

tonikitoo commented on services/ui/ws/server_window.cc:
```diff
@@ -189,10 +189,10 @@ void ServerWindow::SetBoundsFromHostWindowManager(
   gfx::Rect old_bounds = bounds_;
   SetBoundsInternal(bounds, local_surface_id);
   if (bounds.origin() != old_bounds.origin()) {
-	// The root_'s origin mustn't be changed here, so that events are correctly
```
in theory, this should go as a fixup of the patch that introduced it, so that we could squash them, and move commits around independently.

msisov replied:
reverted

tonikitoo reviewed (approved):
lgtm with nits

msisov commented:
done!
